packages/modules/create-imodel/src/components/context/imodel-context.tsx

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React, { createContext, useState } from "react";
+
+import { iModelExtent } from "../../types";
+
+type iModelProps = {
+  name: string;
+  description: string;
+  thumbnail?: { src?: ArrayBuffer; type: string };
+  extent?: iModelExtent | null;
+};
+
+type ContextProps = {
+  iModel: iModelProps;
+  onChange: (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => void;
+  onImageChange: (src: ArrayBuffer, type: string) => void;
+  setImodel: React.Dispatch<
+    React.SetStateAction<{
+      name: string;
+      description: string;
+      thumbnail?:
+        | {
+            src?: ArrayBuffer | undefined;
+            type: string;
+          }
+        | undefined;
+      extent?: iModelExtent | null | undefined;
+    }>
+  >;
+  isThumbnailChanged: boolean;
+};
+
+type userContextProviderPorps = {
+  children: React.ReactNode;
+};
+
+export const IModelContext = createContext<ContextProps>({} as ContextProps);
+
+export const IModelContextProvider = ({
+  children,
+}: userContextProviderPorps) => {
+  const [iModel, setImodel] = useState<{
+    name: string;
```

Comment:
so use `iModelProps` type you have defined above.
also, per naming convention it should be `setIModel` i guess

Reply:
not doing anything on the context now, as per our discussion